modules/miscutil/lib/recommender.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Loads Recommendations stored in Redis."""
+
+import json
+
+from invenio.bibfield import get_record
+from invenio.config import CFG_BASE_URL, \
+    CFG_RECOMMENDER_REDIS, \
+    CFG_RECOMMENDER_PREFIX, \
+    CFG_SITE_RECORD
+from invenio.webuser import collect_user_info
+
+
+_RECOMMENDATION_REDIS = None
+
+
+def get_recommended_records(recid, user_id, maximum=3):
+    """
+    Record recommendations.
+
+    @param recid: Record id for the recommendation.
+    @param threshold: Not used but there for compatibility.
+    @return: List of recommended records [{
+                                          'number': ,
+                                          'record_url': ,
+                                          'record_title': ,
+                                          'record_authors': ,
+                                          }, ]
+    """
+    global _RECOMMENDATION_REDIS
```

Comment:
I would split this file into two, one taking care of the initialization of the recommender, and another one providing the API.